Zabbix update

untergeek opened the pull request:
Full rewrite of the Zabbix output plugin, using the ZabbixProtocol Ruby gem instead of the zabbix_sender binary via the `shellwords` gem.

untergeek committed:
Full rewrite of Zabbix output plugin

This plugin now uses a dedicated Ruby Gem (ZabbixProtocol) rather
than using zabbix_sender via the shell.

This supports integration testing with a real Zabbix Server (v2.2.2)
via a Docker image.

untergeek committed:
Add integration testing instructions

The testing instructions for Docker are here.

untergeek committed:
Update to look cleaner

And remove an unused dev dependency

untergeek committed:
Removed leftover pry bindings

untergeek committed:
Updates for Docker testing

1. Catch when Docker runs locally (unix:///var/run/docker.sock)
2. Ensure Zabbix is actually responding before tests

wiibaa commented on lib/logstash/outputs/zabbix.rb:
```diff
@@ -1,147 +1,185 @@
-#         # only process events with this tag
-#         tags => "zabbix-sender"
+# Each "host" is an identifier, and each item is associated with that host.
+# Item are typed on the Zabbix side.  You can send numbers as strings and Zabbix
```
Item**s** are

wiibaa commented on lib/logstash/outputs/zabbix.rb:
```diff
@@ -1,147 +1,185 @@
+  config :zabbix_value, :validate => :string, :default => "message"
+
+  # The number of seconds to wait before giving up on a connection to the Zabbix
+  # server.  This number should be very small, otherwise delays in delivery of
```
one extra space before **This**

wiibaa commented:
big :+1: for the first docker-enabled tests I see, will try to get inspiration from it. Thx

wiibaa commented:
@untergeek As you are heavily reviewing zabbix, could you have a quick look to the usecase of https://github.com/elastic/logstash-contrib/pull/118 please

untergeek committed:
Doc fixes per @wiibaa

untergeek commented:
Thanks for the doc fixes and the ref to the [PR to put the timestamp in](https://github.com/elastic/logstash-contrib/pull/118).  I'm sure I can do this, if the ZabbixProtocol Rubygem supports it.

untergeek committed:
Test cleanup and feature addition

Implement https://github.com/elastic/logstash-contrib/pull/118 in
a way that makes sense to this plugin.

Add tests for the timestamp changes

Clean up the tests and make the let() statements do their job

Randomize the container names a bit to prevent collisions in the
event of a hung test.

untergeek committed:
Add docker files and documentation

Describe how the image was created and provide the files.

talevy commented on DEVELOPER.md:
```diff
@@ -0,0 +1,35 @@
+
+If you are using Docker already, you can prime your Docker instance with
+
+    docker pull untergeek/logstash_output_zabbix_rspec
```
maybe make this more explicit like this:

```
docker pull untergeek/logstash_output_zabbix_rspec:zabbix_v2.2.2
```

my docker complained that it could not find an image with the "latest" tag (assuming it looks for that by default).

untergeek replied:
Addressed this in 78817b6

untergeek committed:
Update with proper image tag

The docker tag was omitted.  In addition to improving the documentation
These changes make it easier to address future tag changes.

untergeek committed:
Update image tag here also

And also add a "future" link to the docker/README.md file.

talevy commented:
LGTM

suyograo commented:
LGTM
